core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -63,10 +63,12 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
    */
   def writeFully(channel: WritableByteChannel): Unit = {
     for (bytes <- getChunks()) {
-      while (bytes.remaining() > 0) {
+      val curChunkLimit = bytes.limit()
+      while (bytes.hasRemaining) {
         val ioSize = Math.min(bytes.remaining(), bufferWriteChunkSize)
         bytes.limit(bytes.position() + ioSize)
         channel.write(bytes)
+        bytes.limit(curChunkLimit)
```

Comment:
I would rewrite this using:
```
try {
	val ioSize = Math.min(bytes.remaining(), bufferWriteChunkSize)
	bytes.limit(bytes.position() + ioSize)
	channel.write(bytes)
} finally {
	bytes.limit(curChunkLimit)
}
```
to be safe.

Reply:
Right. When channel write throw IOException

Reply:
I have commit this modified